web-js/src/hooks/useDraftTodos.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import * as Realm from "realm-web";
+
+export function useDraftTodos() {
+  const [drafts, setDrafts] = React.useState([]);
+
+  const createDraftTodo = () => {
+    const draftTodo = {
+      _id: new Realm.BSON.ObjectID(),
+      summary: "",
+      isComplete: false,
+    };
+    setDrafts((d) => [...d, draftTodo]);
+  };
+
+  const setDraftTodoSummary = (draft, summary) => {
+    setDrafts((oldDrafts) => {
```

Comment:
TIL you can give setState a function to derive the new state from the previous state.  Very cool